<?xml version="1.0" ?>
<launch>
	<arg name="mode" default="default"/>
	<arg name="namespace" default="hugin"/>
	<arg name="debug" default="0"/>
	<arg name="simulation_freq" default="0.01"/>
	<arg name="mbes_sim_threads" default="4"/>
	<arg name="area" default="single_track"/>


    <!--UFO mapping parameters-->
    <arg name="nodelet_manager" default="bathy_mapper_server_nodelet_manager" />
    <arg name="resolution" default="1" />
    <arg name="depth_levels" default="19" />
    <arg name="multithreaded" default="true" />

	<group ns="$(arg namespace)">

		<!-- Hugin model -->
		<param name="robot_description" command="$(find xacro)/xacro.py '$(find hugin_description)/robots/hugin_auv_$(arg mode).urdf.xacro' debug:=$(arg debug) namespace:=$(arg namespace)" />

		<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen">
		<param name="robot_description" value="/$(arg namespace)/robot_description" />
		</node>

		<!-- UFO mapping -->
	    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen">
            <param name="num_worker_threads" value="12" />
	    </node>

		<node pkg="nodelet" type="nodelet" name="bathy_mapper_server_nodelet" args="load bathy_mapper/BathyMapperNodelet $(arg nodelet_manager)" output="log" required="true">
	        <remap from="cloud_in" to="/gt/mbes_pings" />
	        <param name="output_map" value="$(find bathy_mapper)/maps/$(arg area)_map" />
	        <param name="input_map" value="$(find bathy_mapper)/maps/$(arg area)_map" />
	        <param name="multithreaded" value="$(arg multithreaded)" />
	        <param name="resolution" value="$(arg resolution)" />
	        <param name="depth_levels" value="$(arg depth_levels)" />
			<param name="map_cereal" value="/media/torroba18/Seagate Expansion Drive/HuginDatasets/BornoSMaRC201906/2019_06_24_Biosensor/pp/EM2040/localization_datasets/$(arg area)/map.cereal"/>
			<param name="sim_freq" value="$(arg simulation_freq)" />
	        <param name="world_frame" value="world" />
			<param name="map_frame" value="map" />
			<param name="mbes_link" value="$(arg namespace)/mbes_link" />
			<param name="mbes_open_angle" value="1.5708" />
			<param name="num_beams_sim" value="100" />
			<param name="mbes_sim_as" value="/$(arg namespace)/mbes_sim" />				
		</node>

		<!-- Hugin simulation -->
		<node pkg="auv_2_ros" type="mbes_meas_node" name="mbes_meas_node" output="screen">
			<param name="map_cereal" value="/media/torroba18/Seagate Expansion Drive/HuginDatasets/BornoSMaRC201906/2019_06_24_Biosensor/pp/EM2040/localization_datasets/$(arg area)/map.cereal"/>
			<param name="sim_freq" value="$(arg simulation_freq)" />

			<param name="map_pcl" value="/gt/map" />
			<param name="world_frame" value="world" />
			<param name="map_frame" value="map" />
			<param name="mbes_link" value="$(arg namespace)/mbes_link" />
			<param name="mbes_open_angle" value="1.5708" />
			<param name="num_beams_sim" value="255" />
			<param name="num_threads" value="$(arg mbes_sim_threads)" />
		</node>
		
		<node pkg="auv_2_ros" type="auv_2_ros" name="auv_2_ros" output="screen">
			<param name="trajectory_cereal" value="/media/torroba18/Seagate Expansion Drive/HuginDatasets/BornoSMaRC201906/2019_06_24_Biosensor/pp/EM2040/localization_datasets/$(arg area)/trajectory.cereal"/>
			<param name="sim_freq" value="$(arg simulation_freq)" />

			<param name="mbes_pings" value="/gt/mbes_pings" />
			<param name="map_pcl" value="/gt/map" />
			<param name="debug_pings" value="debug/mbes_pings" />
			<param name="sim_pings" value="/sim/mbes_pings" />
			<param name="world_frame" value="world" />
			<param name="map_frame" value="map" />
			<param name="odom_frame" value="odom" />
			<param name="base_link" value="$(arg namespace)/base_link" />
			<param name="mbes_link" value="$(arg namespace)/mbes_link" />
		</node>

	</group>

</launch>
